modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMDataDefinitionConverterImpl.java

```diff
@@ -158,7 +186,59 @@ public String convertDDMFormLayoutDataDefinition(
 						ddmFormLayout
 					).build());
 
-		return ddmFormLayoutSerializerSerializeResponse.getContent();
+		String content = ddmFormLayoutSerializerSerializeResponse.getContent();
+
+		_addDataDefinitionFieldLinks(
+			structureId, structureLayoutId, ddmForm, _getFieldNames(content),
+			groupId);
+
+		return content;
+	}
+
+	private void _addDataDefinitionFieldLinks(
+			long dataDefinitionId, long dataLayoutId, DDMForm ddmForm,
+			List<String> fieldNames, long groupId)
+		throws Exception {
+
+		Map<String, DDMFormField> ddmFormFieldsMap =
+			ddmForm.getDDMFormFieldsMap(true);
+
+		for (String fieldName : fieldNames) {
+			_deDataDefinitionFieldLinkLocalService.addDEDataDefinitionFieldLink(
+				groupId, _portal.getClassNameId(DDMStructureLayout.class),
```

Comment:
Apparently, getClassNameId has some complex logic that can be avoided for the second time if we assign its result to a variable.